src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -540,6 +540,224 @@ TYPED_TEST(inlet_object, implicit_conversion_primitives)
   EXPECT_EQ(arr, expected_arr);
 }
 
+struct QuuxWithFooArray
+{
+  std::unordered_map<int, Foo> arr;
+  bool operator==(const QuuxWithFooArray& other) const
+  {
+    return arr == other.arr;
+  }
+};
+
+template <>
+struct FromInlet<QuuxWithFooArray>
+{
+  QuuxWithFooArray operator()(const axom::inlet::Table& base)
+  {
+    QuuxWithFooArray q;
+    q.arr = base["arr"].get<std::unordered_map<int, Foo>>();
+    return q;
+  }
+};
+
+TYPED_TEST(inlet_object, nested_array_of_struct)
+{
+  std::string testString =
+    "quux = { [0] = { arr = { [0] = { bar = true; baz = false}, "
+    "                         [1] = { bar = false; baz = true} } }, "
+    "         [1] = { arr = { [0] = { bar = false; baz = false}, "
+    "                         [1] = { bar = true; baz = true} } } }";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& quux_table = inlet.addGenericArray("quux");
+  auto& foo_table = quux_table.addGenericArray("arr");
+
+  foo_table.addBool("bar", "bar's description");
+  foo_table.addBool("baz", "baz's description");
+
+  // Contiguous indexing for generality
+  std::unordered_map<int, QuuxWithFooArray> expected_quuxs = {
+    {0, {{{0, {true, false}}, {1, {false, true}}}}},
```

Comment:
That's a lot of curly braces :). Some newlines might help readability.

Reply:
We might be limited by `clang-format` in adding newlines

Reply:
> We might be limited by `clang-format` in adding newlines

`clang-format` doesn't like new lines?

Reply:
I was suggesting that `clang-format` might remove manually introduced extra lines/whitespace.

(It works at the AST level, not at the line of source code level, so source code whitespaces are not considered in its reformatting)

Reply:
well then, today i learned not to get excited about `clang-format`

Reply:
My takeaway is that our codebase formatting is significantly more consistent since moving to `clang-format`.

There are a few cases where the whitespaces had semantic utility (e.g. constructors for a 4x4 matrix), and in those limited cases, we've manually tagged the few lines (via ``/* clang-format off */``   ``/* clang-format on */`` )